services/cd-service/pkg/repository/repository.go

```diff
@@ -633,6 +634,19 @@ func (r *repository) State() *State {
 	return s
 }
 
+func (r *repository) StateAt(oid *git.Oid) (*State, error) {
+	commit, err := r.repository.LookupCommit(oid)
+	if err != nil {
+		return nil, err
+	}
+	return &State{
+		Filesystem:             fs.NewTreeBuildFS(r.repository, commit.TreeId()),
+		Commit:                 commit,
+		BootstrapMode:          r.config.BootstrapMode,
+		EnvironmentConfigsPath: r.config.EnvironmentConfigsPath,
+	}, nil
+}
+
```

Comment:
this is very similar to `buildState()`.
Can we make it so that `buildState` gets the git.Oid as parameter and then decides how to build it? (if nil, same as before, if not nil, do `repository.LookupCommit`.

I think that would be a little nicer structure.
You can also make another function
```
func buildState() {
return buildState(nil);
}
```
so you don't have to change all callers.

Reply:
I thought about this myself but discarded it because buildState has a very specific handlig for when the repository is empty. I did that no. Can you have another look?